app/models/report_fail.rb

```diff
@@ -0,0 +1,74 @@
+# frozen_string_literal: true
+
+# A class to action on failed labware events
+class ReportFail
+  # The following two modules include methods used by a number of rails helpers, such that we can
+  # use them in eg. form_for
+  extend ActiveModel::Naming
+  include ActiveModel::Conversion
+  include ActiveModel::Validations
+
+  attr_reader :failed_labware_barcodes, :user_code, :failure_id, :failure_options, :selected_option,
+              :disabled_options
+
+  validates :failed_labware_barcodes, :user_code, :failure_id, presence: true
+  validates :user, presence: {
+    message: 'could not be found with that swipecard or barcode. '\
+             'You may need to update your swipecard in Sequencescape.'
+  }
+
+  SAMPLE_INTEGRITY = 'Sample integrity upon receipt'
+  QUANT = 'Failure at quantification (<250ug to proceed to library)'
+  LAB_ERROR = 'Lab error (automation failure)'
+
+  def initialize(user_code, failure_id, failed_labware_barcodes)
+    @user_code = user_code.try(:strip)
+    @failure_id = failure_id.try(:strip)
+    @failed_labware_barcodes = (failed_labware_barcodes || []).map(&:strip)
+
+    @failure_options = {
+      SAMPLE_INTEGRITY => 1,
+      QUANT => 2,
+      LAB_ERROR => 3
+    }
+    @selected_option = 1
+    @disabled_options = [2, 3]
+  end
+
+  def persisted?
+    false
+  end
+
+  def user
+    @user ||= User.find_with_barcode_or_swipecard_code(@user_code)
+  end
+
+  # save attempts to perform the actions, and returns true if it was successful
+  # This maintains compatibility with rails
+  def save
+    return false unless valid?
+
+    failed_labware.each do |labware|
+      labware.events.create_labware_failed!(failure_id, user.login)
+      BroadcastEvent::LabwareFailed.create!(
+        seed: labware,
+        user: user,
+        properties: { failure_reason: failure_options.key(failure_id) }
+      )
+    end
```

Comment:
If one of them fails, from what we talked, we catch the exception of the one that fails and put it in the screen for the user